modules/dxp/apps/commerce/commerce-dashboard-web/src/main/java/com/liferay/commerce/dashboard/web/internal/portlet/action/CommerceDashboardForecastConfigurationAction.java

```diff
@@ -74,12 +80,24 @@ public void processAction(
 		ServiceContext serviceContext = ServiceContextFactory.getInstance(
 			actionRequest);
 
-		String[] assetCategoryIds = ArrayUtil.toStringArray(
-			serviceContext.getAssetCategoryIds());
-
 		setPreference(
-			actionRequest, "assetCategoryIds",
-			StringUtil.merge(assetCategoryIds));
+			actionRequest, "assetCategoryExternalReferenceCodes",
+			StringUtil.merge(
+				TransformUtil.transform(
+					ArrayUtil.toStringArray(
+						serviceContext.getAssetCategoryIds()),
+					assetCategoryId -> {
+						AssetCategory assetCategory =
+							_assetCategoryLocalService.fetchAssetCategory(
+								GetterUtil.getLong(assetCategoryId));
```

Comment:
No need to transform to String and then to Long again. Just use Long values here.